GrocifyApp.UI/src/pages/Products/Products.jsx

```diff
@@ -69,8 +73,22 @@ function Products() {
         fetchData();
     }, []);
 
+    const openModal = () => {
+        setIsModalOpen(true);
+    };
+
+    const closeModal = () => {
+        setIsModalOpen(false);
+    };
+
+    const onConfirmProduct = async (sectionId) => {
+        await getProducts(sectionId);
```

Comment:
you need to setSelectedSection to the new sectionId, otherwise you will get the products from the section of the new product, but the selected section will be the old one.